### DISCUSSION TITLE
App Insights for preview environments?

### DISCUSSION THREAD

**Discussion by GaryParr** [Azure/static-web-apps#1329] [Q&A] [answered]
I recently enabled App Insights to try and debug a 500 error with an API endpoint.  I discovered that none of my preview environments send anything to the App Insights logs.  My site is still completely beta and has no users, so I pushed the changes all the way through to production and trigger the 500 error there before it finally landed in the logs.  Is this expected?

**Comment by GaryParr** [comment, marked as answer]
And, haha... OK, I'm an idiot. I didn't notice the instrumentation key had not been added to my dev and test environment app settings when I enabled App Insights.  Whoops!

**Reply by thomasgauvin** [reply]
Thanks for sharing your resolution steps @garyparr, might be helpful for anyone else! Closing this discussion, feel free to reopen
